internal/transportmock/responder.go

```diff
@@ -0,0 +1,111 @@
+package transportmock
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+)
+
+type Responder func(req *http.Request) (*http.Response, error)
+
+func HTTPResponse(status int, header *map[string][]string, body interface{}, cb func(*http.Request)) Responder {
```

Comment:
What do we think about the naming scheme for Responders? Should we prefix/postfix them with `Response`/`Responder`?